### ISSUE TITLE
File Size limit does not work

### ISSUE THREAD

**Issue by DzenDyn** [fastify/fastify-multipart#428]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.13.0

### Plugin version

7.5.0

### Node.js version

18.14.1

### Operating system

Windows

### Operating system version (i.e. 20.04, 11.3, 10)

Windows 11 (wsl2 ubuntu 22.04.2 LTS)

### Description

The FST_REQ_FILE_TOO_LARGE error occurs only 1 time. After a second attempt to download a large file (just over the specified limit), the download is successful, but the file is truncated to the specified limit.
```
const data = await req.file({ limits: { fileSize: 1048576} })

try {
        console.log(data.filename)
        const file = await data.toBuffer()
        console.log(file.byteLength / 1024 / 1024)
        console.log(data.mimetype)
      } catch (err) {
        if (err instanceof fastify.multipartErrors.RequestFileTooLargeError) {
          console.log(err)
          throw new InvalidRequestError('File too large')
        }
        throw err
      }
```

### Steps to Reproduce

1. Send a file more than limit
2. Get error FST_REQ_FILE_TOO_LARGE
3. Send a file again
4. No error, file size = 1048576

### Expected Behavior

Expected error FST_REQ_FILE_TOO_LARGE on each file upload attempt if file is greater than limit

**Comment by mcollina** [comment]
Thanks for reporting!

Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Comment by gurgunday** [comment]
Hey! I’ll be closing the issue because we cannot help without a reproducible example

For me, the limit is working correctly

Feel free to comment the steps to reproduce your situation

**Comment by tiagokriok** [comment]
Hey, guys!

Firstly, sorry, I don't know if I should create an issue or comment here anyway.

I'm having the same problem. I am sending a file with a size of 30 MB. Sometimes the limit error is thrown, but most of the time the file passes and the file size is reduced. I would love to help resolve this issue if possible, but I would possibly need a guide.


my code:
```typescript
import fastifyMultipart from '@fastify/multipart'
import fastify from 'fastify'

async function main() {
  try {
    const app = fastify()

    await app.register(fastifyMultipart, {
      limits: {
        files: 1,
        fileSize: 10 * 1024 * 1024,
      },
    })

    app.post('/', async (request, _) => {
      const data = await request.file()
      console.dir(data, { depth: null })
      const b = await data?.toBuffer()

      console.log(b?.byteLength)
      return {
        success: true,
      }
    })

    await app.listen({ port: 3000 })
  } catch (e) {
    console.error(e)
    process.exit(1)
  }
}

main()

```

my terminal log
```
<ref *1> {
  type: 'file',
  fieldname: 'file',
  filename: 'design-patterns-en.pdf',
  encoding: '7bit',
  mimetype: 'application/pdf',
  file: FileStream {
    _readableState: ReadableState {
      state: 6192,
      highWaterMark: 16384,
      buffer: BufferList {
        head: {
          data: Buffer(65536) [Uint8Array] [
             37,  80,  68,  70,  45,  49,  46,  53,  10,  37, 226, 227,
            207, 211,  10,  49,  32,  48,  32, 111,  98, 106,  32,  10,
             60,  60,  10,  47,  83, 117,  98, 116, 121, 112, 101,  32,
             47,  70, 111, 114, 109,  10,  47,  70, 105, 108, 116, 101,
            114,  32,  47,  70, 108,  97, 116, 101,  68, 101,  99, 111,
            100, 101,  10,  47,  84, 121, 112, 101,  32,  47,  88,  79,
             98, 106, 101,  99, 116,  10,  47,  77,  97, 116, 114, 105,
            120,  32,  91,  49,  32,  48,  32,  48,  32,  49,  32,  48,
             32,  48,  93,  10,
            ... 65436 more items
          ],
          next: null
        },
        tail: {
          data: Buffer(65536) [Uint8Array] [
             37,  80,  68,  70,  45,  49,  46,  53,  10,  37, 226, 227,
            207, 211,  10,  49,  32,  48,  32, 111,  98, 106,  32,  10,
             60,  60,  10,  47,  83, 117,  98, 116, 121, 112, 101,  32,
             47,  70, 111, 114, 109,  10,  47,  70, 105, 108, 116, 101,
            114,  32,  47,  70, 108,  97, 116, 101,  68, 101,  99, 111,
            100, 101,  10,  47,  84, 121, 112, 101,  32,  47,  88,  79,
             98, 106, 101,  99, 116,  10,  47,  77,  97, 116, 114, 105,
            120,  32,  91,  49,  32,  48,  32,  48,  32,  49,  32,  48,
             32,  48,  93,  10,
            ... 65436 more items
          ],
          next: null
        },
        length: 1
      },
      length: 65536,
      pipes: [],
      flowing: null,
      errored: null,
      defaultEncoding: 'utf8',
      awaitDrainWriters: null,
      decoder: null,
      encoding: null,
      [Symbol(kPaused)]: null
    },
    _events: [Object: null prototype] {
      end: [Function (anonymous)],
      limit: [Function (anonymous)]
    },
    _eventsCount: 2,
    _maxListeners: undefined,
    bytesRead: 65536,
    truncated: false,
    _read: [Function (anonymous)],
    [Symbol(kCapture)]: false
  },
  fields: { file: [Circular *1] },
  _buf: null,
  toBuffer: [AsyncFunction: toBuffer]
}
10485760 -> byteLength
```

**Comment by tiagokriok** [comment]
Hey!

I did more testing today and discovered some strange behavior with Postman.

Yesterday, my tests were being done with Postman, and this behavior mentioned in the previous comment occurs. Today I went to perform the test using Insomnia and the limiter worked correctly, always responding to me with `413 Payload Too Large`. So I believe it is a problem with my Postman or with Postman itself.

My OS: PopOS! 22.04
Postman version: 10.21.14
Node: 20.90
Fastify: 4.25.2
@fastify/multipart: 8.1.0
